src/figuresClasses.ts

```diff
@@ -1,11 +1,93 @@
-export interface Figure {}
+type ShapeType = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: string;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: ShapeType;
+
+  color: string;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const s = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.round(s * 100) / 100;
```

Comment:
The 'getArea' function should not accept any parameters as per the checklist. All necessary values should be available as object fields, which they are in this case.